webapp/src/components/Metrics/PerformanceAnalysis.tsx

```diff
@@ -130,6 +129,17 @@ const PerformanceAnalysis: React.FC<Props> = ({ jobId, pipeline }) => {
     );
   };
 
+  const VisualBarPercentage = (value: number, bgColor: string) =>
+    isNaN(value) ? (
+      "--%"
+    ) : (
+      <VisualBar
+        value={formatRatioAsPercentageString(value as number, 1)}
+        width={100 * value || 1}
```

Comment:
I think the `100 * value || 1` should happen inside `VisualBar`, so that this logic is not duplicated. Also, it's currently not obvious that `VisualBar` expects a number from 1 to 100, so I think it would be clearer:
```suggestion
        width={value}
```